src/app/server/JointFabricDatastore.cpp

```diff
@@ -127,185 +128,575 @@ CHIP_ERROR JointFabricDatastore::RemoveNode(NodeId nodeId)
+                    }
+                    mRefreshingNodeId = kUndefinedNodeId;
+                    mRefreshState     = kIdle;
+                }
+
+                // Continue the state machine (will enter kRefreshingEndpoints branch
+                // when successful and process mEndpointsList).
+                (void) ContinueRefresh();
+            }));
+    }
+    break;
+    case kRefreshingEndpoints: {
+        // 3. cycle through mEndpointsList and add them to the endpoint entries
+        for (const auto & endpoint : mEndpointsList.TakeBuffer())
         {
-            if (it->statusEntry.state == Clusters::JointFabricDatastore::DatastoreStateEnum::kPending)
+            auto it = std::find_if(
+                mEndpointEntries.begin(), mEndpointEntries.end(),
+                [this, &endpoint](const Clusters::JointFabricDatastore::Structs::DatastoreEndpointEntryStruct::Type & entry) {
+                    return entry.nodeID == mRefreshingNodeId && entry.endpointID == endpoint.id;
+                });
+            if (it == mEndpointEntries.end())
             {
-                // TODO: add this entry to the actual device. Only mark as committed if successful.
-                // it->statusEntry.state = Clusters::JointFabricDatastore::DatastoreStateEnum::kCommitted;
+                Clusters::JointFabricDatastore::Structs::DatastoreEndpointEntryStruct::Type newEntry;
+                newEntry.endpointID = endpoint.id;
+                newEntry.nodeID     = mRefreshingNodeId;
+                mEndpointEntries.push_back(newEntry);
             }
-            else if (it->statusEntry.state == Clusters::JointFabricDatastore::DatastoreStateEnum::kDeletePending)
+        }
+
+        // Remove EndpointEntries that are not in the mEndpointsList
+        mEndpointEntries.erase(std::remove_if(mEndpointEntries.begin(), mEndpointEntries.end(),
+                                              [&](const auto & entry) {
+                                                  if (entry.nodeID != mRefreshingNodeId)
+                                                  {
+                                                      return false; // Not for this node, don't remove.
+                                                  }
+                                                  // Remove if not found in mEndpointsList.
+                                                  return !std::any_of(
+                                                      mEndpointsList.TakeBuffer().begin(), mEndpointsList.TakeBuffer().end(),
+                                                      [&](const auto & endpoint) { return entry.endpointID == endpoint.id; });
+                                              }),
+                               mEndpointEntries.end());
+
+        for (auto it = mEndpointGroupIDEntries.begin(); it != mEndpointGroupIDEntries.end();)
```

Comment:
please add a comment before this line to make clear the intention

Reply:
Removed the commented block as it was outdated.